### ISSUE TITLE
Problem with concatenate str

### ISSUE THREAD

**Issue by piasekps** [DurgNomis-drol/ha_toyota#119]
**Describe the bug**
While checking the logs I found this error:

```
This error originated from a custom integration.

Logger: custom_components.toyota
Source: custom_components/toyota/__init__.py:105
Integration: Toyota Connected Services (documentation, issues)
First occurred: 12:36:01 PM (1 occurrences)
Last logged: 12:36:01 PM

Unexpected error fetching toyota data: can only concatenate str (not "dict") to str
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/toyota/__init__.py", line 105, in async_update_data
    data = await asyncio.gather(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/client.py", line 419, in get_driving_statistics
    raw_statistics = await self.api.get_driving_statistics_endpoint(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/api.py", line 96, in get_driving_statistics_endpoint
    return await self.controller.request(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/controller.py", line 219, in request
    "Internal server error occurred! - " + response
TypeError: can only concatenate str (not "dict") to str

```
I'm not sure when this happend and how to reproduce it.

**Screenshots**

**Additional context**

**Comment by briandelicata** [comment]
> **Describe the bug** While checking the logs I found this error:
> 
> ```
> This error originated from a custom integration.
> 
> Logger: custom_components.toyota
> Source: custom_components/toyota/__init__.py:105
> Integration: Toyota Connected Services (documentation, issues)
> First occurred: 12:36:01 PM (1 occurrences)
> Last logged: 12:36:01 PM
> 
> Unexpected error fetching toyota data: can only concatenate str (not "dict") to str
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
>     self.data = await self._async_update_data()
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in _async_update_data
>     return await self.update_method()
>   File "/config/custom_components/toyota/__init__.py", line 105, in async_update_data
>     data = await asyncio.gather(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/client.py", line 419, in get_driving_statistics
>     raw_statistics = await self.api.get_driving_statistics_endpoint(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/api.py", line 96, in get_driving_statistics_endpoint
>     return await self.controller.request(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/controller.py", line 219, in request
>     "Internal server error occurred! - " + response
> TypeError: can only concatenate str (not "dict") to str
> ```
> 
> I'm not sure when this happend and how to reproduce it.
> 
> **Screenshots** If applicable, add screenshots to help explain your problem.
> 
> **Additional context** Add any other context about the problem here.

Same issue my end - even with the last version update:

```
022-12-08 14:54:36.641 ERROR (MainThread) [custom_components.toyota] Unexpected error fetching toyota data: can only concatenate str (not "dict") to str
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 225, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 181, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/toyota/__init__.py", line 105, in async_update_data
    data = await asyncio.gather(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/client.py", line 419, in get_driving_statistics
    raw_statistics = await self.api.get_driving_statistics_endpoint(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/api.py", line 96, in get_driving_statistics_endpoint
    return await self.controller.request(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/controller.py", line 219, in request
    "Internal server error occurred! - " + response
TypeError: can only concatenate str (not "dict") to str
```

**Comment by dvejsada** [comment]
> **Describe the bug** While checking the logs I found this error:
> 
> ```
> This error originated from a custom integration.
> 
> Logger: custom_components.toyota
> Source: custom_components/toyota/__init__.py:105
> Integration: Toyota Connected Services (documentation, issues)
> First occurred: 12:36:01 PM (1 occurrences)
> Last logged: 12:36:01 PM
> 
> Unexpected error fetching toyota data: can only concatenate str (not "dict") to str
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
>     self.data = await self._async_update_data()
>   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in _async_update_data
>     return await self.update_method()
>   File "/config/custom_components/toyota/__init__.py", line 105, in async_update_data
>     data = await asyncio.gather(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/client.py", line 419, in get_driving_statistics
>     raw_statistics = await self.api.get_driving_statistics_endpoint(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/api.py", line 96, in get_driving_statistics_endpoint
>     return await self.controller.request(
>   File "/usr/local/lib/python3.10/site-packages/mytoyota/controller.py", line 219, in request
>     "Internal server error occurred! - " + response
> TypeError: can only concatenate str (not "dict") to str
> ```
> 
> I'm not sure when this happend and how to reproduce it.
> 
> **Screenshots** If applicable, add screenshots to help explain your problem.
> 
> **Additional context** Add any other context about the problem here.

Same issue here as well.

**Comment by inspirepfs** [comment]
Logger: custom_components.toyota
Source: custom_components/toyota/__init__.py:105
Integration: Toyota Connected Services ([documentation](https://github.com/DurgNomis-drol/ha_toyota), [issues](https://github.com/DurgNomis-drol/ha_toyota/issues))
First occurred: 16:51:16 (2 occurrences)
Last logged: 16:52:10

Unexpected error fetching toyota data: can only concatenate str (not "dict") to str
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 258, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 217, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/toyota/__init__.py", line 105, in async_update_data
    data = await asyncio.gather(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/client.py", line 419, in get_driving_statistics
    raw_statistics = await self.api.get_driving_statistics_endpoint(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/api.py", line 96, in get_driving_statistics_endpoint
    return await self.controller.request(
  File "/usr/local/lib/python3.10/site-packages/mytoyota/controller.py", line 219, in request
    "Internal server error occurred! - " + response
TypeError: can only concatenate str (not "dict) to str

I also have this issue too any update on resolving this issue. Thanks Peter

**Comment by CM000n** [comment]
We recently released version 2.0.0 of the Custom Component with major changes and support for the new Mytoyota API.
As we unfortunately cannot provide support for Custom Component versions < v2.0.0, I hereby close your issue.
If the issue still applies to the current version, you are of course free to open a new issue. 😊
